vavr/src/main/java/io/vavr/collection/Traversable.java

```diff
@@ -798,12 +797,7 @@ default T last() {
      */
     @SuppressWarnings("unchecked")
     default Option<T> max() {
-        if (isEmpty()) {
-            return Option.none();
-        } else {
-            final Traversable<T> ts = isTraversableAgain() ? this : toStream();
-            return ts.maxBy(Comparators.naturalComparator());
-        }
+        return maxBy(Comparators.naturalComparator());
```

Comment:
* we do not need to wrap a traversable-once into a Stream because no element is touched before reduction
* maxBy API say that the empty case is already covered.